### ISSUE TITLE
Eden Style Governance Milestone Ticket

### ISSUE THREAD

**Issue by n13** [hypha-dao/dho-web-client#1536]
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by sceenius** [comment]
Playing the above through in condensed form, this is similar to a role application without knowing yet who will be assigned to the role - a process decides that vs. a vote on an applicant. 

1. Create Eden Proposal
2. Apply for Eden Proposal (optional - submit video or select time)
3. Launch Eden event round ED - set timer, group applicants (randomize)
3.1 For each group: watch/meet, then vote
4. Launch Eden event round CD - set timer, group applicants (randomize)
4.1 For each group: watch/meet, then vote
5. Launch Eden event round HD - set timer
5.1 Final group watches/meets, then votes
6. Close Eden event - announce HD, assign HD to proposal as winner (owner)

With ED = Eden Delegate, CD = Chief Delegate, HD = Head Delegate

Couple of notes:

* this is another flavor of Liquid Democracy (LD), the diff here is that it happens for each and every proposal, LD makes the delegates permanent, but only if there are "domain experts". There is a question if Eden proposals really need real-time meetings or if this can be handled async (via video recordings) to save everyone's time. The most complex UI element (to me) is the vote button that needs to be shown on applicants from your current group only (you don't see any others, kind of a filter). Assigning voice and badges is a nice side benefit, but the main concern is to find the HD for the Eden proposal.

**Comment by n13** [comment]
> Playing the above through in condensed form, this is similar to a role application without knowing yet who will be assigned to the role - a process decides that vs. a vote on an already known applicant.
> 
> 1. Create Eden Proposal
> 2. Apply for Eden Proposal (optional - submit video or select time)
> 3. Launch Eden event round ED - set timer, group applicants (randomize)
>    3.1 For each group: watch/meet, then vote
> 4. Launch Eden event round CD - set timer, group applicants (randomize)
>    4.1 For each group: watch/meet, then vote
> 5. Launch Eden event round HD - set timer
>    5.1 Final group watches/meets, then votes
> 6. Close Eden event - announce HD, assign HD to proposal as winner (owner)
> 
> With ED = Eden Delegate, CD = Chief Delegate, HD = Head Delegate
> 
> Couple of notes:
> 
> This is another form of Liquid Democracy (LD), the diff here is that it happens for each and every proposal and there is only 1 delegate, LD makes the delegates permanent, but only if there are "domain experts". There is a question if Eden proposals really need real-time meetings or if this can be handled async (via video recordings) to save everyone's time. The most complex UI element (to me) is the vote button that needs to be shown on applicants from your current group only (you don't see any others where you are not part of the selected group, it's kind of a filter). Assigning voice and badges is a nice side benefit, but the main concern is to find the HD for the Eden proposal.
> 
> ![image](https://user-images.githubusercontent.com/24725582/194171168-7052da31-e3c8-4cc4-a715-e260d72fcd35.png)

There is an existing process and we will need to recreate it.

There were several surprises, details, and unexpected problems in this process.

If we design a new process, then we will come across different details, surprises and unexpected problems

So I don't think it is feasible to deviate from the current process at all

In addition, we need to show the existing EOS on Eden process in our UI as part of one of the milestones, and that will have to be exactly like their process anyway. 

So yeah - I have some screenshots of the existing process I can upload later.

**Comment by n13** [comment]
I actually think the existing Eden UI is good as far as function is concerned. 

What makes it confusing is the visual layout which is not very good. We can improve on that. 

For each round there's 2 viiews

- An overview - this shows all groups, all members, and all votes in real time
- A group view - for someone who is in a group, there's a detail view just for that group, where I can hit the vote button on any of the other members.

**Comment by sceenius** [comment]
The  EDEN democratic process elects a new leadership team (delegates) that will vote on new proposals as long as that role or badge is active. So, I think it is not used as a voting method for each new proposal, but rather as a way to delegate the vote to the elected lead until that role/badge expires.  So, 

1. EDEN democratic process is launched and winners are recorded on chain (via UI/Dashboard or https://genesis.eden.eoscommunity.org/)
2. New proposals that are created in the DAO can select the DAO consensus model as a voting method (need to add the voting method as option)
3. The only person(s) allowed to vote on that proposal are the elected delegates (or only the Head Delegate, not sure).
4. Voting proceeds as usual and the proposal is either accepted or rejected (by the delegates).

**Comment by sceenius** [comment]
Further inquiry to the Eden folks (10/27)

> Another question I have - once the Eden election is complete and you have new delegates, you want to bring their power into the DAO. My thinking is to design the DAO consensus model in a way that either limits votes to the delegates (fairly easy to do, can adjust to the elected level of HD, CD or L1) or gives delegates a voice boost (more involved, since we need to take that away once the delegate status has ended plus we'd have to calculate how much additional voice power is needed to effectively overwrite non-delegates). 

My initial opinion is that limiting the Voting Power to the delegates ( with adjusted levels to their respective ranks ) is the preferred way.  It means to me that being a delegate of any level comes with a  “privilege” AND a responsibility (!) to vote/participate. The quarterly Eden elections would be a chance to achieve this “privilege”. Designing it this way would potentially drive more meaningful engagement and dedicated participation to both processes. 

**My reflection on this:**
Very good alignment on the idea to give delegates special status in the DAO. I am no thinking to assign them special privileges in the community doughnut, such as being able to create a proposal inside the nucleus (in addition to the full voting rights in the community space).  That way we don't have to struggle with voice power.

**Comment by sceenius** [comment]
322332The one badge I see being valuable for Eden is that of a "delegate" which can work with the Eden election process for this milestone. Sketching some requirements for that smart badge - 

(1) This badge is only active between two elections and is removed/inactive otherwise
(2) Badge holders have the full voting power IF the Eden consensus method has been selected (under vote settings)
(3) Possible voice multipliers can apply that give (former) delegates more voice after they step down from the role
(4) Possible delegate levels are shown on the badge (3=L1, 2=CD, 1=HD, 0=alumni)

**Comment by sceenius** [comment]
Current summary of all requirements gathered so far:

1) Eden election process - no change in the protocol, results are mapped to the DAO, Eden bot guides participants
2) Eden consensus method - new voting method in the DAO, delegates "run the show" (in the community and/or nucleus)
3) Eden delegate badges - new smart badges for delegates, different levels are accounted for

**Comment by sceenius** [comment]
The corresponding features we can implement are

1. Eden election process - add new delegates to DAO (EOS account ID + badge)
2. Eden consensus method - configure new voting method in DAO (Eden election ID)
3. Eden delegate badges - add smart badges to DAO (voice replacement)

**Comment by sceenius** [comment]
Stories:

As an Eden community member, I can vote with voice =1 on any proposal
As an L1 delegate, I am eligible for a L1 badge and can vote with voice = 2 on any proposal
As a Chief Delegate, I am eligible for a CD  badge and can vote with voice = 4 on any proposal
As a Head Delegate, I am eligible for a HD badge and can vote with voice = 16 on any proposal
As an administrator, I can switch the DAO to the "Eden Consensus" method; this will only be enabled if all required badges are active
As a core member, I can create new proposals

**Comment by sceenius** [comment]
On Monday Dec 12 at 2pm ET the Eden folks are testing the new Eden Up Vote Bot, I will try to be there to learn more about it. This is just another indicator that we should NOT replace or rebuild  what they are doing, instead just "import" the results into the DAO. They are making continuous updates to the election process.

![image](https://user-images.githubusercontent.com/24725582/206801241-7f87a02e-d144-4a0a-9e18-204d424c98d9.png)

**Comment by sceenius** [comment]
Selected Eden Bot screenshots:

![image](https://user-images.githubusercontent.com/24725582/207138465-e7f9c3ec-737f-437d-b145-14509724c8dd.png)
![image](https://user-images.githubusercontent.com/24725582/207138549-dba98467-92a6-42b5-9490-f54c3995b5b2.png)
![image](https://user-images.githubusercontent.com/24725582/207138594-8b72f30e-6cee-4d62-98da-5be501f1bfb2.png)

**Comment by sceenius** [comment]
The Eden DAO could be an embedded view on the right menu bar - 
![image](https://user-images.githubusercontent.com/24725582/207142433-52aef7b3-2846-4b9a-80bd-86e300623530.png)

https://genesis.eden.eoscommunity.org/election

**Comment by sceenius** [comment]
Update from Eden:

Chief delegates are stored on chain, others are only in the members table (not sure if we can distinguish) - 

CDs:
https://bloks.io/account/genesis.eden?loadContract=true&tab=Tables&table=elect.state&account=genesis.eden&scope=&limit=100

Members:
https://bloks.io/account/genesis.eden?loadContract=true&tab=Tables&table=member&account=genesis.eden&scope=&limit=1000

**Comment by sceenius** [comment]
Full write-up of the recipe is here - https://notepad.hypha.earth/_CtQVhzbS-q5UD_UXtTNTw?view
